sdk/storage/Azure.Storage.Blobs/src/BlobExtensions.cs

```diff
@@ -0,0 +1,311 @@
+                EncryptionKeySha256 = properties.EncryptionKeySha256,
+                EncryptionScope = properties.EncryptionScope,
+                AccessTier = properties.AccessTier,
+                AccessTierInferred = properties.AccessTierInferred,
+                ArchiveStatus = properties.ArchiveStatus,
+                AccessTierChangedOn = properties.AccessTierChangedOn,
+                VersionId = properties.VersionId,
+                IsLatestVersion = properties.IsCurrentVersion,
+                TagCount = properties.TagCount,
+                ExpiresOn = properties.ExpiresOn,
+                IsSealed = properties.IsSealed,
+                RehydratePriority = properties.RehydratePriority
+            };
+
+        /// <summary>
+        /// Internal. Parses Object Replication Policy ID from Rule ID and sets the Policy ID for source blobs.
+        /// </summary>
+        /// <param name="OrIds">
+        /// Unparsed Object Replication headers.
+        /// For source blobs, the dictionary will contain keys that contain the policy id and rule id separated
+        /// by a underscore (e.g. policyId_ruleId). The value of these keys will be the replication status (e.g. Complete, Failed).
+        /// For destination blobs, the dictionary will contain one entry where the key will be "policy-id"
+        /// and the value will be the destination policy id. No parsing will be required for this.
+        /// </param>
+        /// <returns>
+        /// If the blob has object replication policy(s) applied and is the source blob, this method will return a
+        /// List of <see cref="ObjectReplicationPolicy"/>, which contains the Policy ID and the respective
+        /// rule(s) and replication status(s) for each policy.
+        /// If the blob has object replication policy applied and is the destination blob,
+        /// this method will return default as the policy id should be set in ObjectReplicationDestinationPolicyId
+        /// (e.g. <see cref="BlobProperties.ObjectReplicationDestinationPolicyId"/>,<see cref="BlobDownloadDetails.ObjectReplicationDestinationPolicyId"/>).
+        /// </returns>
+        internal static IList<ObjectReplicationPolicy> ParseObjectReplicationIds(this IDictionary<string, string> OrIds)
+        {
+            // If the dictionary contains a key with policy id, we are not required to do any parsing since
+            // the policy id should already be stored in the ObjectReplicationDestinationPolicyId.
+            if (OrIds.First().Key == "policy-id")
+            {
+                return default;
+            }
+            List <ObjectReplicationPolicy> OrProperties = new List<ObjectReplicationPolicy>();
+            foreach (KeyValuePair<string, string> status in OrIds)
+            {
+                string[] ParsedIds = status.Key.Split('_');
+                int policyIndex = OrProperties.FindIndex(policy => policy.PolicyId == ParsedIds[0]);
+                if (policyIndex > -1)
```

Comment:
Nit - variables should be camelCased like `parsedIds` and we should also check the length before indexing

Reply:
Fixed camel case.  @amnguye